### ISSUE TITLE
Previous in-app notifications about curation status get updated to latest curation status

### ISSUE THREAD

**Issue by lobbikerivm** [IQSS/dataverse#10835]
**What steps does it take to reproduce the issue?**
I start a curation process, and the status is changed multipletimes. I receive notifications in-app

* When does this issue occur?
after every new notification 

* Which page(s) does it occurs on?
User> Notifications

* What happens?
Previous notifications get updated. The text of all the notification mentions the most recent curation status only

* To whom does it occur (all users, curators, superusers)?
superusers, didn't test other users
* What did you expect to happen?
Previous notifications should remain intact, so a user can see when the curation label is changed to what?

**Which version of Dataverse are you using?**
6.3

**Any related open or closed issues to this bug report?**
didn't see any

**Screenshots:**
![curation_notification](https://github.com/user-attachments/assets/a2fa3db0-16ca-4f7a-8ed2-b507243030e4)

**Are you thinking about creating a pull request for this issue?**
Help is always welcome, is this bug something you or your organization plan to fix?
I am not able to, not familiar with Java

**Comment by lobbikerivm** [comment]
Yes, the are custom curation labels we use. I get a notification every time the curation label changes. Older in-app notifications for the same dataset get overwritten, so the all show the latest curation label (see the timestamps in the screenshot)

**Comment by qqmyers** [comment]
FWIW: The existing design generates messages from the current db state, so it makes sense that all the curation state messages incorrectly show the current state. Since the initial development, an 'additionalInfo' field has been added to the UserNotification class - that could probably be used to store the  at the time the notification was created.

**Comment by lobbikerivm** [comment]
I took a look in the database, and notice that the related thing stored in table usernotification is the objectid, which relates to the datasetversionid. In datasetversion is the current curation label stored. so the information about history of curation labels is not in the database. For a published dataset version the label is NULL, and the corresponding notification turns to 'The status of dataset sdsdf has been updated to .', as you would expect. Correct @qqmyers, the field additionalinfo in the database table is empty, and seems like a nice place to store the status at time of notification.

Thanks for having a look, guys.

Stefan

**Comment by pdurbin** [comment]
Are we thinking this is a new bug as of 6.3? Or does it go back several versions?

**Comment by qqmyers** [comment]
It's the way it has always worked/not worked. FWIW: The same thing is true for other messages, e.g. if you change the name of a dataset, the new name will show in all old messages about it, etc. Something like that may be OK, but its possible there are other messages that get confusing if you make changes later on.

**Comment by pdurbin** [comment]
Oooooh. I understand now. I agree this behavior is a bit odd. And yeah, I suppose the solution is to store more data in the database. I haven't looked at `additionalInfo` but, sure, it's probably worth a shot to store it there.
